graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/parser/PipelineRuleParser.java

```diff
@@ -758,6 +758,15 @@ public void exitOr(RuleLangParser.OrContext ctx) {
             checkBinaryExpression(ctx);
         }
 
+        @Override
+        public void exitNot(RuleLangParser.NotContext ctx) {
+            final Expression expression = parseContext.expressions().get(ctx.expression());
+            Class type = expression.getType();
+            if (!Boolean.class.equals(type)) {
```

Comment:
I might be nitpicking here, but wouldn't it be more correct to use:
```suggestion
            if (!Boolean.class.isAssignableFrom(type)) {
```

Reply:
We could do that but Boolean is a final class, so it's effectively identical :)